typecheck/typecheck.go

```diff
@@ -1738,6 +1738,14 @@ func (c *Checker) exprPartial(e expr.Expr, hint typeHint) (p partial) {
 		if right.mode == modeInvalid {
 			return right
 		}
+		// handle possible comma-ok
```

Comment:
Problem is, the comma-ok is the exceptional case, not the norm. What happens with:

```
ch := make(chan int, 1)
ch <- 1
func f(x int) { print(x) }
f(<-ch)
```

?

Similarly, `print(*(<-ch))` for `chan *int`, or many other examples.

Instead of returning a tuple, Assign needs to work out that the expression on the other side is a channel (or a map), and infer the ok case.

Reply:
ok.
so... can I add a field `CommaOK` to `stmt.Assign`:

```go
type Assign struct {
        Decl  bool
        CommaOK  bool
        Left  []expr.Expr
        Right []expr.Expr // TODO: give up on multiple rhs values for now.
}
```

the typechecker would set it to true when needed.
it would help during evaluation of the statement later on

Reply:
Yes that would be fine. Similarly the type checker sets the `ElideError` bool in expr.Call.